content/en/docs/concepts/services-networking/service.md

```diff
@@ -91,32 +91,32 @@ spec:
     targetPort: 9376
 ```
 
-This specification will create a new Service object named “my-service” which
-targets TCP port 9376 on any Pod with the `"app=MyApp"` label.
+This specification creates a new Service object named “my-service”, which
+targets TCP port 9376 on any Pod with the `app=MyApp` label.
 
-This Service will also be assigned an IP address (sometimes called the "cluster IP"),
-which is used by the service proxies
+This Service also assigns an IP address (sometimes called the "cluster IP"),
```

Comment:
The Service doesn't do the assigning.

Suggestion: This Service is assigned an IP address.
OR
Kubernetes assigns this Service and IP address.

Reply:
Behind the scenes, what's the controller doing the assigning?

AIUI sometimes it's something outside Kubernetes that assigns the Cluster IP. Have I got that right?